test/chisel.py

```diff
@@ -0,0 +1,178 @@
+    client.agree_to_tos(account)
+    return client
+
+def get_chall(client, domain):
+    """Ask the server for an authz, return the authz and an HTTP-01 challenge."""
+    authz = client.request_domain_challenges(domain)
+    for chall_body in authz.body.challenges:
+        if isinstance(chall_body.chall, challenges.HTTP01):
+            return authz, chall_body
+    raise "No HTTP-01 challenge found"
+
+def make_authzs(client, domains):
+    """Make authzs for each of the given domains. Return a list of authzs
+       and challenges."""
+    authzs, challenges = [], []
+    for d in domains:
+        authz, chall_body = get_chall(client, d)
+
+        authzs.append(authz)
+        challenges.append(chall_body)
+    return authzs, challenges
+
+class ValidationError(Exception):
+    """An error that occurs during challenge validation."""
+    def __init__(self, domain, problem_type, detail, *args, **kwargs):
+        self.domain = domain
+        self.problem_type = problem_type
+        self.detail = detail
+
+    def __str__(self):
+        return "%s: %s: %s" % (self.domain, self.problem_type, self.detail)
+
+def issue(client, authzs, cert_output=None):
+    """Given a list of authzs that are being processed by the server,
+       wait for them to be ready, then request issuance of a cert with a random
+       key for the given domains."""
+    domains = [authz.body.identifier.value for authz in authzs]
+    pkey = OpenSSL.crypto.PKey()
+    pkey.generate_key(OpenSSL.crypto.TYPE_RSA, 2048)
+    csr = OpenSSL.crypto.X509Req()
+    csr.add_extensions([
+        OpenSSL.crypto.X509Extension(
+            'subjectAltName',
+            critical=False,
+            value=', '.join('DNS:' + d for d in domains).encode()
+        ),
+    ])
+    csr.set_pubkey(pkey)
+    csr.set_version(2)
+    csr.sign(pkey, 'sha256')
+
+    cert_resource = None
+    try:
+        cert_resource, _ = client.poll_and_request_issuance(jose.ComparableX509(csr), authzs)
+    except acme_errors.PollError as error:
+        # If we get a PollError, pick the first failed authz and turn it into a more
+        # useful ValidationError that contains details we can look for in tests.
+        for authz in error.updated:
+            updated_authz = json.loads(urllib2.urlopen(authz.uri).read())
+            domain = authz.body.identifier.value,
+            for c in updated_authz['challenges']:
+                if 'error' in c:
+                    err = c['error']
+                    raise ValidationError(domain, err['type'], err['detail'])
+        # If none of the authz's had an error, just re-raise.
+        raise
+    if cert_output != None:
+        pem = OpenSSL.crypto.dump_certificate(OpenSSL.crypto.FILETYPE_PEM,
+                                              cert_resource.body)
+        with open(cert_output, 'w') as f:
+            f.write(pem)
+    return cert_resource
+
+def http_01_answer(client, chall_body):
```

Comment:
The approach I took here was to only provide the "build a client" default in functions where it would be useful. For instance, http_01_answer is only ever called from contexts that already have a client.

I figure if we need that functionality later, we can add it, but until then let's follow the "You Ain't Gonna Need It" philosophy.